### PR TITLE
Adding render tests for image fallback with feature constant expressions and `addImage` operation

### PR DESCRIPTION
**Pull request by SnailBones**
As pointed out by @alexshalamov, the tests added in #11049 and #11359 contain no feature constant expressions, but are all using `["image", ["get", "icon"]]`. So in this PR I add two tests that use constant values, i.e. `["image", "fav-bicycle-18"]` 

Since data and constant expressions are evaluated in different codepaths, these additional tests should improve test coverage in JS and [Native](https://github.com/mapbox/mapbox-gl-native-internal/pull/2655) and ensure that image fallback expressions work in all expected conditions.

I also add two tests ensuring image fallback works with sprites added through `addImage` (as opposed to images in the spritesheet) and make a small improvement to organization.

## Launch Checklist

 - [X] briefly describe the changes in this PR
 - [X] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

**Comment by SnailBones** [comment]
Added two more tests: `/add-image` and `feature-state/add-image` to test with the `addImage` operation. GL JS tests fail unless the layer is moved to `addLayer`, this suggests that the map doesn'trerender on `addImage`.

This is not an issue in Native, where both these tests pass, as do variants without using `addLayer`.

These tests pass even before [this fix](https://github.com/mapbox/mapbox-gl-native-internal/pull/2655/commits/3400ba5ecc829838e88f6d3d200f0076be7d21fb) and without triggering a breakpoint in the second `VertexVectors` overload [here](https://github.com/mapbox/mapbox-gl-native-internal/blob/3400ba5ecc829838e88f6d3d200f0076be7d21fb/src/mbgl/renderer/paint_property_binder.hpp#L799).

File: test/integration/render-tests/image-fallback-nested/feature-state/add-image/style.json [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        [
+          "addImage",
+          "rocket",
+          "./image/rocket.png"
+        ],
+                [
+          "wait"
+        ],
+        [
+          "addLayer",
```
Comment (alexshalamov) [inline test/integration/render-tests/image-fallback-nested/feature-state/add-image/style.json:18]: addLayer will trigger tile re-layout. Would it be possible to move layer definition to style definition? List of operations may be:

1. wait
2. add image
3. set feature state

Comment (SnailBones) [reply test/integration/render-tests/image-fallback-nested/feature-state/add-image/style.json:18]: Thank you @alexshalamov!

I tried the test with the layer definition in the style definition as you describe (in place of the current  `"layers": []`. This causes the test to pass in GL Native but fail in GL JS. I think this is from an issue unique to render tests in GL JS.

I see two ways around this:
- resolve this issue in GL JS to get consistent render test behavior
- add the test in GL Native only (and perhaps open an issue to address this in GL JS at a later date).

I think the latter in the simplest approach for now, what do you think?

Comment (SnailBones) [reply test/integration/render-tests/image-fallback-nested/feature-state/add-image/style.json:18]: I just tested out the changes you suggest, and they still don't seem to use the intended [updateVertexVectors](https://github.com/mapbox/mapbox-gl-native-internal/blob/3400ba5ecc829838e88f6d3d200f0076be7d21fb/src/mbgl/renderer/paint_property_binder.hpp#L799) function in Native.

**Review by karimnaaji — APPROVED** [review]
Per chat with @SnailBones approving this PR as even though it's not a strict requirement for gl-native, it still has a fair amount of render test coverage increase.

**Pull request merged**